src/Core/Managed/Shared/TelemetryClient.cs

```diff
@@ -379,11 +379,12 @@ public void Initialize(ITelemetry telemetry)
 
             if (string.IsNullOrEmpty(instrumentationKey))
             {
-                instrumentationKey = PlatformSingleton.Current.GetEnvironmentVariable(InstrumentationKeyWebSitesEnvironmentVariable);
-                
-                if (string.IsNullOrEmpty(instrumentationKey))
+                // If no configuration override was passed to the constructor then the default configuration instance will have been used.
+                // If no iKey was present in the config then the environment variable will have been loaded into the default instance.
+                instrumentationKey = this.configuration.InstrumentationKey;
+                if (string.IsNullOrEmpty(instrumentationKey) && this.configuration != TelemetryConfiguration.Active)
```

Comment:
I think we need this logic because if you specify a custom config with no iKey it won't pickup from the environment variable but this way provides a fallback when trying to send telemetry.